// vetur-vls/lib/services/formatterOptions.js
(function (factory) {
    if (typeof module === "object" && typeof module.exports === "object") {
        var v = factory(require, exports);
        if (v !== undefined) module.exports = v;
    }
    else if (typeof define === "function" && define.amd) {
        define(["require", "exports"], factory);
    }
})(function (require, exports) {
    "use strict";
    Object.defineProperty(exports, "__esModule", { value: true });
    var defaultHtmlOptions = {
        brace_style: 'collapse',
        end_with_newline: false,
        indent_char: ' ',
        indent_handlebars: false,
        indent_inner_html: false,
        indent_scripts: 'keep',
        indent_size: 2,
        indent_with_tabs: false,
        max_preserve_newlines: 1,
        preserve_newlines: true,
        unformatted: [],
        wrap_line_length: 0,
        wrap_attributes: 'force-aligned' // Wrap attributes to new lines [auto|force|force-aligned|force-expand-multiline] ["auto"]
    };
    exports.defaultHtmlOptions = defaultHtmlOptions;
    var defaultCssOptions = {
        end_with_newline: false,
        indent_char: ' ',
        indent_size: 2,
        indent_with_tabs: false,
        newline_between_rules: true,
        preserve_newlines: true,
        selector_separator: ' ',
        selector_separator_newline: true // Separate selectors with newline or not (e.g. 'a,\nbr' or 'a, br')
    };
    exports.defaultCssOptions = defaultCssOptions;
});
