// vetur-vls/lib/services/vueCompletion.js
(function (factory) {
    if (typeof module === "object" && typeof module.exports === "object") {
        var v = factory(require, exports);
        if (v !== undefined) module.exports = v;
    }
    else if (typeof define === "function" && define.amd) {
        define(["require", "exports", "vscode-languageserver-types"], factory);
    }
})(function (require, exports) {
    "use strict";
    Object.defineProperty(exports, "__esModule", { value: true });
    var vscode_languageserver_types_1 = require("vscode-languageserver-types");
    function doVueComplete() {
        var topLevelCompletions = [
            {
                label: 'scaffold',
                documentation: 'Scaffold <template>, <script> and <style>',
                kind: vscode_languageserver_types_1.CompletionItemKind.Snippet,
                insertTextFormat: vscode_languageserver_types_1.InsertTextFormat.Snippet,
                insertText: "<template>\n  ${}\n</template>\n\n<script>\nexport default {\n\n}\n</script>\n\n<style>\n\n</style>\n"
            },
            {
                label: 'template with html',
                documentation: 'Scaffold <template> with html',
                kind: vscode_languageserver_types_1.CompletionItemKind.Snippet,
                insertTextFormat: vscode_languageserver_types_1.InsertTextFormat.Snippet,
                insertText: "<template>\n  ${}\n</template>\n"
            },
            {
                label: 'template with pug',
                documentation: 'Scaffold <template> with pug',
                kind: vscode_languageserver_types_1.CompletionItemKind.Snippet,
                insertTextFormat: vscode_languageserver_types_1.InsertTextFormat.Snippet,
                insertText: "<template lang=\"pug\">\n  ${}\n</template>\n"
            },
            {
                label: 'script with JavaScript',
                documentation: 'Scaffold <script> with JavaScript',
                kind: vscode_languageserver_types_1.CompletionItemKind.Snippet,
                insertTextFormat: vscode_languageserver_types_1.InsertTextFormat.Snippet,
                insertText: "<script>\nexport default {\n  ${}\n}\n</script>\n"
            },
            {
                label: 'script with TypeScript',
                documentation: 'Scaffold <script> with TypeScript',
                kind: vscode_languageserver_types_1.CompletionItemKind.Snippet,
                insertTextFormat: vscode_languageserver_types_1.InsertTextFormat.Snippet,
                insertText: "<script lang=\"ts\">\nexport default {\n  ${}\n}\n</script>\n"
            },
            {
                label: 'style with CSS',
                documentation: 'Scaffold <style> with CSS',
                kind: vscode_languageserver_types_1.CompletionItemKind.Snippet,
                insertTextFormat: vscode_languageserver_types_1.InsertTextFormat.Snippet,
                insertText: "<style>\n${}\n</style>\n"
            },
            {
                label: 'style with scss',
                documentation: 'Scaffold <style> with scss',
                kind: vscode_languageserver_types_1.CompletionItemKind.Snippet,
                insertTextFormat: vscode_languageserver_types_1.InsertTextFormat.Snippet,
                insertText: "<style lang=\"scss\">\n${}\n</style>\n"
            },
            {
                label: 'style with less',
                documentation: 'Scaffold <style> with less',
                kind: vscode_languageserver_types_1.CompletionItemKind.Snippet,
                insertTextFormat: vscode_languageserver_types_1.InsertTextFormat.Snippet,
                insertText: "<style lang=\"less\">\n${}\n</style>\n"
            },
            {
                label: 'style with sass',
                documentation: 'Scaffold <style> with sass',
                kind: vscode_languageserver_types_1.CompletionItemKind.Snippet,
                insertTextFormat: vscode_languageserver_types_1.InsertTextFormat.Snippet,
                insertText: "<style lang=\"sass\">\n${}\n</style>\n"
            },
            {
                label: 'style with postcss',
                documentation: 'Scaffold <style> with postcss',
                kind: vscode_languageserver_types_1.CompletionItemKind.Snippet,
                insertTextFormat: vscode_languageserver_types_1.InsertTextFormat.Snippet,
                insertText: "<style lang=\"postcss\">\n${}\n</style>\n"
            },
            {
                label: 'style with stylus',
                documentation: 'Scaffold <style> with stylus',
                kind: vscode_languageserver_types_1.CompletionItemKind.Snippet,
                insertTextFormat: vscode_languageserver_types_1.InsertTextFormat.Snippet,
                insertText: "<style lang=\"stylus\">\n${}\n</style>\n"
            },
        ];
        return {
            isIncomplete: false,
            items: topLevelCompletions
        };
    }
    exports.doVueComplete = doVueComplete;
});
